test.js

```diff
@@ -391,3 +391,23 @@ it('missing source maps should also be cached', function(done) {
     '1', // The retrieval should only be attempted once
   ]);
 });
+
+it('finds source maps with charset specified', function() {
+  var sourceMap = createMultiLineSourceMap()
+  var source = [ 'throw new Error("test");' ];
+  var expected = [
+    'Error: test',
+    /^    at Object\.exports\.test \(.*\/line1\.js:1001:101\)$/
+  ];
+
+  fs.writeFileSync('.generated.js', 'exports.test = function() {' +
+    source.join('\n') + '};//@ sourceMappingURL=data:application/json;charset=utf8;base64,' +
+    new Buffer(sourceMap.toString()).toString('base64'));
+  try {
+    delete require.cache[require.resolve('./.generated')];
+    require('./.generated').test();
+  } catch (e) {
+    compareLines(e.stack.split('\n'), expected);
+  }
+  fs.unlinkSync('.generated.js');
```

Comment:
Fair enough.

Would you mind adding a small comment above the test explaining that this is some duplicated code and why it is reasonable to do so?